Flaky test: test_dashboard_scheduling_diagnostic races scheduler feedback

## Summary

`test_dashboard_scheduling_diagnostic` is flaky because it checks `pending_reason` immediately after job submission, before the scheduler has had a chance to evaluate the job. When the scheduler is slow, the reason is still the generic `"Pending scheduler feedback"` instead of the expected CPU-specific message.

**Failed run:** https://github.com/marin-community/marin/actions/runs/23078555218/job/67043359016?pr=3651
(attempt 1 failed, attempt 2 passed — classic flake)

## Failure

```
tests/e2e/test_smoke.py:421: in test_dashboard_scheduling_diagnostic
    assert "cpu" in status.pending_reason.lower()
E   assert 'cpu' in 'pending scheduler feedback'
```

The job requests 999,999 CPUs (intentionally unschedulable). The test expects the `pending_reason` to mention "cpu" (e.g., "Insufficient cpu capacity"), but the scheduler hasn't run yet so the reason is still the initial `"Pending scheduler feedback"`.

## Root cause

In `test_smoke.py:418-421`:

```python
with smoke_cluster.launched_job(TestJobs.quick, "smoke-diag-cpu", cpu=999_999) as job:
    status = smoke_cluster.status(job)
    assert status.state == cluster_pb2.JOB_STATE_PENDING
    assert "cpu" in status.pending_reason.lower()
```

`status()` is called once with no retry. The scheduler runs asynchronously, so on a slow CI runner the scheduling diagnostic hasn't been computed yet.

## Suggested fix

Poll until `pending_reason` contains `"cpu"` (with a timeout), similar to how `wait_for_state` works:

```python
with smoke_cluster.launched_job(TestJobs.quick, "smoke-diag-cpu", cpu=999_999) as job:
    deadline = time.monotonic() + smoke_cluster.job_timeout
    while time.monotonic() < deadline:
        status = smoke_cluster.status(job)
        if "cpu" in status.pending_reason.lower():
            break
        time.sleep(0.2)
    assert status.state == cluster_pb2.JOB_STATE_PENDING
    assert "cpu" in status.pending_reason.lower()
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/lib/iris/tests/e2e/test_smoke.py
+++ b/lib/iris/tests/e2e/test_smoke.py
@@ -416,7 +416,14 @@ def test_dashboard_scheduling_diagnostic(smoke_cluster, smoke_page, smoke_screen
     """Scheduling diagnostic shows pending reason for oversized job."""
     smoke_cluster.wait_for_workers(1, timeout=smoke_cluster.job_timeout)
     with smoke_cluster.launched_job(TestJobs.quick, "smoke-diag-cpu", cpu=999_999) as job:
-        status = smoke_cluster.status(job)
+        # Poll until the scheduler has evaluated the job and produced a
+        # CPU-specific pending reason (avoids racing the scheduler loop).
+        deadline = time.monotonic() + smoke_cluster.job_timeout
+        while time.monotonic() < deadline:
+            status = smoke_cluster.status(job)
+            if "cpu" in status.pending_reason.lower():
+                break
+            time.sleep(0.2)
         assert status.state == cluster_pb2.JOB_STATE_PENDING
         assert "cpu" in status.pending_reason.lower()
 

